src/paging.rs

```diff
@@ -685,6 +685,13 @@ impl<T: Translation> PageTableWithLevel<T> {
             if level == LEAF_LEVEL {
                 // Put down a page mapping.
                 entry.set(pa, flags | Attributes::TABLE_OR_PAGE);
+            } else if !entry.is_table_or_page()
+                && entry.flags() == flags
+                && entry.output_address()
+                    == PhysicalAddress(align_down(chunk.0.start.0, granularity))
```

Comment:
This check for the output address isn't quite right. Suppose someone tries to map VA 0x1000 to PA 0 or 0x2000 when there is already a block mapping which maps the block starting at 0 to 0 (and hence 0x1000 is mapped to 0x1000), this will return true when it should return false.